src/Java.Interop.Tools.JavaTypeSystem/Adapters/ManagedApiImporter.cs

```diff
@@ -0,0 +1,379 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Java.Interop.Tools.Cecil;
+using Java.Interop.Tools.JavaTypeSystem.Models;
+using Mono.Cecil;
+using Mono.Collections.Generic;
+
+namespace Java.Interop.Tools.JavaTypeSystem
+{
+	public static class ManagedApiImporter
+	{
+		public static JavaTypeCollection Parse (AssemblyDefinition assembly, JavaTypeCollection collection)
+		{
+			var types_to_add = new List<JavaTypeModel> ();
+
+			foreach (var md in assembly.Modules)
+				foreach (var td in md.Types) {
+					if (!ShouldSkipType (td) && ParseType (td, collection) is JavaTypeModel type)
+						types_to_add.Add (type);
+				}
+
+			// This needs to be done ordered from least nested to most nested, in order for nesting to work.
+			// That is, 'android.foo.Blah' needs to be added before 'android.foo.Blah.Bar'.
+			// Plus, we may have unnested managed types that are actually nested in Java-land:
+			// ex: IContextMenu and IContextMenuContextMenuItem
+			foreach (var type in types_to_add.OrderBy (t => t.FullName.Count (c => c == '.')).ToArray ())
+				AddReferenceTypeRecursive (type, collection);
+
+			return collection;
+		}
+
+		public static JavaTypeModel? ParseType (TypeDefinition type, JavaTypeCollection collection)
+		{
+			if (!type.IsPublic && !type.IsNested)
+				return null;
+
+			if (!ShouldImport (type))
+				return null;
+
+			var model = type.IsInterface ? (JavaTypeModel?) ParseInterface (type, collection) : ParseClass (type, collection);
+
+			if (model is null)
+				return null;
+
+			foreach (var nested in type.NestedTypes)
+				if (ParseType (nested, collection) is JavaTypeModel nested_model)
+					model.NestedTypes.Add (nested_model);
+
+			return model;
+		}
+
+		static bool ShouldImport (TypeDefinition td)
+		{
+			// We want to exclude "IBlahInvoker" and "IBlahImplementor" and "BlahConsts" types
+			if (td.Name.EndsWith ("Invoker")) {
```

Comment:
[`string.EndsWith(string)`](https://docs.microsoft.com/en-us/dotnet/api/system.string.endswith?view=net-5.0#System_String_EndsWith_System_String_) is locale-aware.  This should instead use [`string.EndsWith(string, StringComparison.Ordinal)`](https://docs.microsoft.com/en-us/dotnet/api/system.string.endswith?view=net-5.0#System_String_EndsWith_System_String_System_StringComparison_).

Ditto other uses of `.EndsWith()`.

Reply:
I rebased on `main`, which now raises errors for these issues, and fixed them.